src/app/components/crypto-assets/crypto-currency-asset/crypto-currency-asset-item.tsx

```diff
@@ -1,21 +1,63 @@
-import { StackProps } from '@stacks/ui';
+import { useState } from 'react';
+import { toast } from 'react-hot-toast';
+
+import { StackProps, useClipboard } from '@stacks/ui';
 import { forwardRefWithAs } from '@stacks/ui-core';
 
 import type { AllCryptoCurrencyAssetBalances } from '@shared/models/crypto-asset-balance.model';
 
+import { useAnalytics } from '@app/common/hooks/analytics/use-analytics';
+
 import { CryptoCurrencyAssetItemLayout } from './crypto-currency-asset-item.layout';
 
 interface CryptoCurrencyAssetItemProps extends StackProps {
   assetBalance: AllCryptoCurrencyAssetBalances;
   assetSubBalance?: AllCryptoCurrencyAssetBalances;
   icon: JSX.Element;
+  usdBalance?: string;
+  address?: string;
   isPressable?: boolean;
+  canCopy?: boolean;
 }
 export const CryptoCurrencyAssetItem = forwardRefWithAs(
   (props: CryptoCurrencyAssetItemProps, ref) => {
-    const { assetBalance, assetSubBalance, icon, isPressable, ...rest } = props;
+    const {
+      assetBalance,
+      assetSubBalance,
+      icon,
+      isPressable,
+      address,
+      canCopy,
+      usdBalance,
+      ...rest
+    } = props;
     const { balance, asset } = assetBalance;
+    const [isHovered, setIsHovered] = useState(false);
+    const { onCopy, hasCopied } = useClipboard(address || '');
+    const analytics = useAnalytics();
+
+    function onHover() {
+      if (!canCopy) {
```

Comment:
probably its better to put this logic in `balance-list.tsx` component?

Reply:
which logic specifically? the hover state?

Reply:
yes